test_api.py

```diff
@@ -880,126 +882,56 @@ def test_create_host_with_invalid_ansible_host(self):
 
                 self.verify_error_response(error_host, expected_title="Bad Request")
 
-    def test_create_host_with_20_byte_MAC_address(self):
-        system_profile = {
-            "network_interfaces": [{"mac_address": "00:11:22:33:44:55:66:77:88:99:aa:bb:cc:dd:ee:ff:00:11:22:33"}]
-        }
+    def test_create_host_with_tags_doesnt_work(self):
+        host_data = HostWrapper(test_data(tags=[{"namespace": "ns", "key": "some_key", "value": "val"}]))
+        create_response = self.post(HOST_URL, [host_data.data()], 207)
+        host_id = create_response["data"][0]["host"]["id"]
+        response = self.get(f"{HOST_URL}/{host_id}/tags")
 
-        host_data = HostWrapper(test_data(system_profile=system_profile))
+        self.maxDiff = None
+        self.assertEqual(response["results"][host_id], [])
 
-        response = self.post(HOST_URL, [host_data.data()], 207)
+    def test_update_host_with_tags_doesnt_change_tags(self):
+        host_data = HostWrapper(
+            test_data(tags=[{"namespace": "ns", "key": "some_key", "value": "val"}], fqdn="fqdn", id=generate_uuid())
+        )
 
-        self._verify_host_status(response, 0, 201)
+        message = {
+            "operation": "add_host",
+            "data": {
+                "id": host_data.id,
+                "account": host_data.account,
+                "display_name": host_data.display_name,
+                "tags": host_data.tags,
+                "reporter": host_data.reporter,
+                "stale_timestamp": host_data.stale_timestamp,
+                "facts": host_data.facts,
+                "fqdn": host_data.fqdn,
+            },
+        }
 
-        created_host = self._pluck_host_from_response(response, 0)
+        with self.app.app_context():
+            mock_event_producer = Mock()
+            handle_message(json.dumps(message), mock_event_producer)
+            json.loads(mock_event_producer.write_event.call_args[0][0])
 
-        original_id = created_host["id"]
+        # attempt to update
+        self.post(HOST_URL, [host_data.data()], 207)
 
-        host_lookup_results = self.get(f"{HOST_URL}/{original_id}", 200)
+        tags_after_update = self.get(f"{HOST_URL}/{host_data.id}/tags")
 
-        self._validate_host(host_lookup_results["results"][0], host_data, expected_id=original_id)
+        # check the tags haven't updated
+        self.assertNotEqual(host_data.tags, tags_after_update)
```

Comment:
This doesn’t test the right thing. _tags_after_update_ contains a full get tags response that should look like this:

```python
{'total': 1, 'count': 1, 'page': 1, 'per_page': 50, 'results': {'eeaa3525-fddc-49cc-8145-c1620c7976c4': [{'namespace': 'ns', 'key': 'some_key', 'value': 'val'}]}}
```

We need to test that the tags were not updated. So we need to compare `tags_after_update["results"][host_id]["tags"]` to the original tags.

```suggestion
        self.assertEqual(host_data.tags, tags_response["results"][host_id])
```